python/pyarrow/tests/test_compute.py

```diff
@@ -1859,35 +1859,42 @@ def validate_select_k(select_k_indices, arr, order, stable_sort=False):
             assert actual == expected
 
     arr = pa.array([1, 2, None, 0])
-    for order in ["descending", "ascending"]:
-        for k in [0, 2, 4]:
+    for k in [0, 2, 4]:
```

Comment:
For the record, what happens if `k` is greater than the input length?

Reply:
when `k` is greater than input length, `k = input.length`. Same thing happens when there are less non-null elements in that array `k = first_null_index` after PartitionNull.